### PR TITLE
Reach scenarios  1 2 3a

### PR DESCRIPTION
**Pull request by my2457**
Add Reach scenarios 1,2,3a

### CODE REVIEW

File: src/evaluations/data/evaluation_configs.py [hunk 1]
```diff
@@ -96,15 +96,70 @@ def _smoke_test(num_runs=NUM_RUNS_VALUE):
...
+  as total. Othersize, we take it as scenario 2 with remarketing size as
+  int(universe_size*remarketing_rate)
+
+  See scenario 1 / 2:
+  Independent m-publishers / n-publishers independently serve a remarketing list
+  https://github.com/world-federation-of-advertisers/cardinality_estimation_evaluation_framework/blob/master/doc/cardinality_and_frequency_estimation_evaluation_framework.md#scenario-1-independence-m-publishers-with-homogeneous-user-reach-probability
+  Returns:
+    A list of ScenarioConfigs of scenario 1 / 2 with selected parameters.
+  """
+  name_to_choices_of_set_size_list = {
```
Comment (my2457) [inline src/evaluations/data/evaluation_configs.py:116]: Seeing this set size setup in different scenarios, I'm thinking maybe better to merge them together in a single method for all scenarios 1,2,3,5? let me know if that makes sense.

File: src/evaluations/data/evaluation_configs.py [hunk 2]
```diff
@@ -115,9 +170,10 @@ def _generate_configs_scenario_3b(universe_size, num_sets, small_set_size,
     user_activity_assciation: user activity association used in the Exponential
       Bow model. Should be one of the defined user activity association defined
       by the set_generator.USER_ACTIVITY_ASSOCIATION_XXX.
-
+      For 3(a) user_activity_assciation=set_generator.USER_ACTIVITY_ASSOCIATION_XXX
```
Comment (huangxichen1) [inline src/evaluations/data/evaluation_configs.py:173]: Do you mean USER_ACTIVITY_ASSOCIATION_INDEPENDENT?

File: src/evaluations/data/evaluation_configs.py (hunk 1 above)
Comment (huangxichen1) [reply src/evaluations/data/evaluation_configs.py:116]: Makes sense to me.

File: src/evaluations/data/evaluation_configs.py (hunk 2 above)
Comment (my2457) [reply src/evaluations/data/evaluation_configs.py:173]: Nice catch!

File: src/evaluations/data/evaluation_configs.py (hunk 1 above)
Comment (kungfucraig) [reply src/evaluations/data/evaluation_configs.py:116]: +1

Comment (my2457) [reply src/evaluations/data/evaluation_configs.py]: Added, I also updated scenario 1 to include `gradually_smaller`. There are more scenarios in 5 for different combinations of large/small set, but I didn't add it to default considering 5 is more sensitive to order and those are probably specifically designed for 5, but please let me know if it makes sense. Thanks!

File: src/evaluations/data/evaluation_configs.py [hunk 3]
```diff
@@ -262,24 +320,21 @@ def _generate_configs_scenario_5(num_sets, small_set_size,
   Returns:
     A list of ScenarioConfigs of scenario 5 Sequentiall correlated sets.
   """
+
   name_to_choices_of_set_size_list = {
-      '1st_small_then_large': [small_set_size] + [large_set_size] * (
-          num_sets - 1),
+      **_get_default_name_to_choices_of_set_size_list(
```
Comment (my2457) [inline src/evaluations/data/evaluation_configs.py:325]: I didn't add all of the fields to default method considering this scenario is sensitive to orders and we may want to test more cases that may not be needed in other scenarios, but let me know if that's the case

File: src/evaluations/data/tests/evaluation_configs_test.py [hunk 4]
```diff
@@ -29,43 +29,77 @@ class EvaluationConfigTest(absltest.TestCase):
...
-    conf_list = evaluation_configs._generate_configs_scenario_3b(
-        universe_size=200,
-        num_sets=3,
-        small_set_size=50,
-        large_set_size=100,
-        user_activity_assciation=(set_generator
-                                  .USER_ACTIVITY_ASSOCIATION_IDENTICAL)
-    )
+  def test_generate_configs_scenario_1_2_set_sizes_correct(self):
+    for remarketing_rate in [None, 0.2]:
```
Comment (kungfucraig) [inline src/evaluations/data/tests/evaluation_configs_test.py:33]: It'd be better to use a parameterized test here. Makes test debugging easier... less whack-a-mole type debugging.

File: src/evaluations/data/tests/evaluation_configs_test.py [hunk 5]
```diff
@@ -29,43 +29,77 @@ class EvaluationConfigTest(absltest.TestCase):
...
-        'exponential_bow-user_activity_association:identical-'
-        'universe_size:200-small_set:50-large_set:100-'
-        'set_type:small_then_last_large': [48, 48, 84],
-        'exponential_bow-user_activity_association:identical-'
-        'universe_size:200-small_set:50-large_set:100-'
-        'set_type:gradually_smaller': [84, 67, 55]
-    }
+      universe_size
+      expected = {
+        f'{type_header}-universe_size:2000-small_set:50-large_set:100-set_type:1st_half_small_2nd_half_large': [50, 100, 100],
```
Comment (kungfucraig) [inline src/evaluations/data/tests/evaluation_configs_test.py:55]: 80 char line length please

File: src/evaluations/data/tests/evaluation_configs_test.py (hunk 4 above)
Comment (my2457) [reply src/evaluations/data/tests/evaluation_configs_test.py:33]: Was following the existing code logic, let me refactor a little bit

**Pull request merged**